### ISSUE TITLE
Using an API endpoint as a connectivity URL

### ISSUE THREAD

**Issue by vin047** [rwbutler/Connectivity#9]
Hi, do the custom URLs have to serve a HTML file, or is an API endpoint (GET endpoint that returns the word "Success") sufficient?

**Comment by rwbutler** [comment]
The current implementation only checks that the word 'Success' is returned in the response so that would work.

I meant to get around to writing a more robust implementation of this check just in case it ever turns out to be the case that captive portal happens to contain the word as part of the web page served however I'll ensure that I introduce this in a backwards-compatible manner when I do. Hope this helps!

**Comment by rwbutler** [comment]
I've added the option to customise the validity test used when checking connectivity URLs. To only check that the response contains the word `Success` you can set the following:

```
connectivity.expectedResponseString = "Success" // this is the default value
connectivity.validationMode = .containsExpectedResponseString
```

**Comment by vin047** [comment]
Ah thats brilliant, thanks!

**Comment by vin047** [comment]
Hey @rwbutler, is there any way of being able to set an authorization header token for a connectivity URL? I have an API endpoint thats protected using JWT tokens. Currently have to run a separate unprotected route for checking connectivity status - just wondering if there's an easy way around this, as it would be good to allow only authorised clients to access the endpoint.

**Comment by rwbutler** [comment]
I haven’t got a way of doing this just yet but I’ll look into it and get back to you 👍🏻

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by rwbutler** [comment]
@vin047 I've just released v2.2.0 which includes a `bearerToken` property. If you set your JWT token here it should be supplied to your endpoint. Hope this helps.

**Comment by vin047** [comment]
Hey, sorry for the delay in getting back to you. I just had a chance to implement version 2.2.0 and can confirm the bearerToken feature works as expected. Thanks a lot for the effort and for this library! 👍 🥇
